src/Assignments/ProblemIframe.tsx

```diff
@@ -82,50 +163,33 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
             // console.error('Hijacker: Could not find the form to insert the listener');
             return;
         }
-        problemForm.addEventListener('submit', (event: { preventDefault: () => void; }) => {
+
+        problemForm.addEventListener('submit', _.debounce((event: { preventDefault: () => void; }) => {
             event.preventDefault();
             if (_.isNil(problemForm)) {
                 console.error('Hijacker: Could not find the form when submitting the form');
                 setError('An error occurred');
                 return;
             }
-            let formData = new FormData(problemForm);
             let clickedButton = problemForm.querySelector('.btn-clicked') as HTMLButtonElement;
             if (_.isNil(clickedButton)) {
                 setError('Hijacker: An error occurred');
                 console.error('Could not find the button that submitted the form');
                 return;
             }
-            formData.set(clickedButton.name, clickedButton.value);
-            const submiturl = problemForm.getAttribute('action');
-            if(_.isNil(submiturl)) {
+            console.log('preparing formdata and submitting!');
```

Comment:
I personally prefer not to have console logs in there

Reply:
Oops, missed a cleanup item